tests/UnitTests/CoreGraphics.drawing/CGContextDrawing_PatternTests.cpp

```diff
@@ -0,0 +1,373 @@
+        int size = ::testing::get<2>(GetParam());
+
+        return CFStringCreateWithFormat(nullptr,
+                                        nullptr,
+                                        CFSTR("TestImage.CGContext.CGPatternStencil.%s.%s.%d.png"),
+                                        stencil.name,
+                                        stencilColor.name,
+                                        size);
+    }
+};
+
+TEST_P(CGPatternStencil, DrawStencils) {
+    CGPatternDrawPatternCallback callback = ::testing::get<0>(GetParam()).callback;
+    const CGFloat* stencilColor = ::testing::get<1>(GetParam()).color;
+    int size = ::testing::get<2>(GetParam());
+
+    CGContextRef context = GetDrawingContext();
+    CGRect bounds = GetDrawingBounds();
+
+    static const CGPatternCallbacks callbacks = { 0, callback, NULL };
+
+    CGColorSpaceRef baseSpace = CGColorSpaceCreateDeviceRGB();
+    CGColorSpaceRef patternSpace = CGColorSpaceCreatePattern(baseSpace);
+    CGContextSetFillColorSpace(context, patternSpace);
+    CGColorSpaceRelease(patternSpace);
+    CGColorSpaceRelease(baseSpace);
+
+    CGPatternRef pattern = CGPatternCreate(
+        NULL, CGRectMake(0, 0, size, size), CGAffineTransformIdentity, size, size, kCGPatternTilingConstantSpacing, false, &callbacks);
+
+    CGContextSetFillPattern(context, pattern, stencilColor);
+    CGPatternRelease(pattern);
+    CGContextFillRect(context, bounds);
+}
+
+INSTANTIATE_TEST_CASE_P(CGPatternTests,
+                        CGPatternStencil,
+                        ::testing::Combine(::testing::ValuesIn(stencilPatternCallback),
+                                           ::testing::ValuesIn(stencilColors),
+                                           ::testing::ValuesIn(patternsizes)));
+
+#pragma endregion Stencil Pattern
+
+#pragma region Colored Pattern
+
+void drawColoredPatternMultiSquare(void* info, CGContextRef context) {
+    CGFloat subunit = 5; // the pattern cell itself is 16 by 18
+
+    CGRect myRect1 = { { 0, 0 }, { subunit, subunit } }, myRect2 = { { subunit, subunit }, { subunit, subunit } },
+           myRect3 = { { 0, subunit }, { subunit, subunit } }, myRect4 = { { subunit, 0 }, { subunit, subunit } };
+
+    CGContextSetRGBFillColor(context, 0, 0, 1, 0.5);
+    CGContextFillRect(context, myRect1);
+    CGContextSetRGBFillColor(context, 1, 0, 0, 0.5);
+    CGContextFillRect(context, myRect2);
+    CGContextSetRGBFillColor(context, 0, 1, 0, 0.5);
+    CGContextFillRect(context, myRect3);
+    CGContextSetRGBFillColor(context, .5, 0, .5, 0.5);
+    CGContextFillRect(context, myRect4);
+}
+
+PatternCallback ColoredPatternCallback[] = { { "drawColoredPatternMultiSquare", &drawColoredPatternMultiSquare } };
+
+//<PatternCallback,alpha,pattern size>
+class CGPatternColored : public WhiteBackgroundTest<>,
+                         public ::testing::WithParamInterface<::testing::tuple<PatternCallback, CGFloat, int>> {
```

Comment:
super duper nit: should use CGFloat rather than int for CGSize width, height values